client/my-sites/stats/features/modules/stats-locations/stats-locations.tsx

```diff
@@ -102,15 +105,25 @@ const StatsLocations: React.FC< StatsModuleLocationsProps > = ( { query, summary
 	const geoMode = GEO_MODES[ selectedOption ];
 	const title = optionLabels[ selectedOption ]?.selectLabel;
 
+	const { supportsLocationsStats: supportsLocationsStatsFeature } = useSelector( ( state ) =>
+		getEnvStatsFeatureSupportChecks( state, siteId )
+	);
+
 	// Main location data query
 	const {
-		data = [],
+		data: locationsViewsData = [],
 		isLoading: isRequestingData,
 		isError,
 	} = useLocationViewsQuery< StatsLocationViewsData >( siteId, geoMode, query, countryFilter, {
-		enabled: ! shouldGate,
+		enabled: ! shouldGate && supportsLocationsStatsFeature,
 	} );
 
+	const countriesViewsData = useSelector( ( state ) =>
+		getSiteStatsNormalizedData( state, siteId, statType, query )
+	) as [ id: number, label: string ];
```

Comment:
Tried to check just one or the other, but there were complaints that hooks cannot be conditionally rendered. Shouldn't be an actual performance issue, given the `StatsLocationViewsData` has the `enabled` flag and `getSiteStatsNormalizedData` makes use of cache.

Reply:
Sounds good to me. I assume the `getSiteStatsNormalizedData` selector retrieves stats from the Redux state, which are stored thanks to `QuerySiteStats`. Maybe it’s time to render that query component only when needed? Otherwise, we’ll be attempting to fetch the legacy country-views API even when we don’t need it.

Just to clarify, this isn’t an issue related to your PR. It could be addressed in another one if you prefer.

Reply:
I'm all in for further improving that. Would you mind opening an issue about it? I'm not sure I got 100% of the context to better describe what we need to do.

Reply:
Sure, I'll take a look and open the issue in the project board :)